bootstrap/pkg/kfapp/coordinator/coordinator.go

```diff
@@ -63,9 +63,28 @@ func GetKfApp(kfdef *kfdefsv2.KfDef) kftypes.KfApp {
 	return _coordinator
 }
 
-func getConfigFromCache(pathDir string, kfDef *kfdefsv2.KfDef) ([]byte, error) {
+func getStaticConfigForPlatform(kfDef *kfdefsv2.KfDef) string {
 
-	configPath := filepath.Join(pathDir, kftypes.DefaultConfigDir)
+	return ""
+	// Platform specific options to override config.
+	// eg a platform might want to use a static config file
+}
+
+func getConfigFromCache(kfDef *kfdefsv2.KfDef, staticConfigFile string) ([]byte, error) {
+
+	// Check if a static config file has been specified
+	if staticConfigFile != "" {
+		configFileBytes, err := ioutil.ReadFile(staticConfigFile)
+		if err != nil {
+			return nil, &kfapis.KfError{
+				Code:    int(kfapis.INTERNAL_ERROR),
+				Message: fmt.Sprintf("error reading specified config file %s: %v", staticConfigFile, err),
+			}
+		}
+		return configFileBytes, nil
+	}
+
+	configPath := filepath.Join(kfDef.Spec.Repo, kftypes.DefaultConfigDir)
```

Comment:
It looks like we are looking at a KfDef spec in order to locate a YAML file from which we get the KfDef spec.

Can we simplify this and not use a KfDef spec to provide whatever info (e.g. repo information to locate the yaml file).

For example suppose the syntax was 

```
kfctl init --config=file://myrepo/path/kfdef.yaml
```

Then the argument fully specifies the location so there's no reason to reference some other kfdef spec to figure out where to locate it from.

Reply:
I agree that it's a convoluted way to do it and that we should refactor this part of kfctl to be more simple/clear in what it's doing.
However, I  believe it is a bit out of scope for this PR and would be a pretty big PR on its own.